client/state/ui/editor/actions.js

```diff
@@ -28,3 +31,27 @@ export function toggleEditorDraftsVisible() {
 		type: EDITOR_SHOW_DRAFTS_TOGGLE
 	};
 }
+
+/**
+ * Returns an action object used in signalling that the media modal current
+ * view should be updated in the context of the post editor.
+ *
+ * @param  {ModalViews} view Media view
+ * @return {Object}          Action object
+ */
+export function setEditorMediaModalView( view ) {
+	let stat;
+	switch ( view ) {
+		case ModalViews.LIST: stat = 'view_list'; break;
+		case ModalViews.DETAIL: stat = 'view_detail'; break;
+		case ModalViews.GALLERY: stat = 'view_gallery'; break;
+		case ModalViews.IMAGE_EDITOR: stat = 'view_edit'; break;
+	}
```

Comment:
this seems like a strange change of coding style. what if we converted this into an immutable assignment to stat with a good default?

```js
// static mapping defined in module scope outside of function
const viewMapping = {
	[ ModalViews.LIST ]: 'view_list',
	[ ModalViews.DETAIL ]: 'view_detail',
	[ ModalViews.GALLERY ]: 'view_gallery',
	[ ModalViews.IMAGE_EDITOR ]: 'view_edit',
};

const stat = get( viewMapping, view );
const action = setMediaModalView( view );

return stat
	? withAnalytics( bumpStat( 'editor_media_actions', stat ), action )
	: action;
```